### PR TITLE
Adding a filter for state in product review grid

### PR DESCRIPTION
**Pull request by mamazu**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13                  |
| Bug fix?        | no                                                       |
| New feature?    | yes                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no |
| Related tickets | -                      |
| License         | MIT                                                          |

## Use case
When the admin wants to see which product reviews are still pending and needs to be reviewed, he needs to filter for the product reviews.

### CODE REVIEW

File: src/Sylius/Behat/Page/Admin/ProductReview/IndexPage.php [hunk 1]
```diff
@@ -37,4 +42,12 @@ private function getActionButtonsField(array $parameters): NodeElement
...
         return $tableAccessor->getFieldFromRow($table, $row, 'actions');
     }
+
+    protected function getDefinedElements(): array
+    {
+        return parent::getDefinedElements() + [
+            'filter_state' => '#criteria_status',
+        ];
+    }
+
```
Comment (lchrusciel) [inline src/Sylius/Behat/Page/Admin/ProductReview/IndexPage.php]:
```suggestion
```

File: features/product/managing_product_reviews/filtering_product_reviews.feature [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        And I am logged in as an administrator
+
+    @ui
+    Scenario: Browsing accepted reviews
+        When I want to browse product reviews
+        And I choose "accepted" as a status filter
+        And I filter
+        Then I should see a single product review in the list
+        And I should see the product review "Awesome" in the list
+
```
Comment (lchrusciel) [inline features/product/managing_product_reviews/filtering_product_reviews.feature]:
```suggestion
```

**Review by lchrusciel — APPROVED** [review]

**Comment by loic425** [comment]
@Florian-Merle Sth to implement on your PR for grids refactoring on Sylius :)

**Review by loic425 — APPROVED** [review]

File: features/product/managing_product_reviews/filtering_product_reviews.feature [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        Given the store has customer "Mike Ross" with email "ross@teammike.com"
+        And the store has a product "Lamborghini Gallardo Model"
+        And this product has a review titled "Awesome" and rated 5 with a comment "Nice product" added by customer "ross@teammike.com"
+        And this product has a new review titled "Bad" and rated 1 added by customer "ross@teammike.com"
+        And I am logged in as an administrator
+
+    @ui
+    Scenario: Browsing accepted reviews
+        When I want to browse product reviews
+        And I choose "accepted" as a status filter
```
Comment (loic425) [inline features/product/managing_product_reviews/filtering_product_reviews.feature]: Maybe "Accepted"?

Comment (mamazu) [reply features/product/managing_product_reviews/filtering_product_reviews.feature]: Should work as well. Let's try.

Comment (mamazu) [reply features/product/managing_product_reviews/filtering_product_reviews.feature]: Done

**Comment by mamazu** [comment]
@lchrusciel Could I get a final review?

File: features/product/managing_product_reviews/filtering_product_reviews.feature [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+Feature: Browsing product reviews
+    In order to work with multiple reviews
+    As an Administrator
+    I want to filter product reviews
+
+    Background:
+        Given the store has customer "Mike Ross" with email "ross@teammike.com"
+        And the store has a product "Lamborghini Gallardo Model"
+        And this product has a review titled "Awesome" and rated 5 with a comment "Nice product" added by customer "ross@teammike.com"
+        And this product has a new review titled "Bad" and rated 1 added by customer "ross@teammike.com"
```
Comment (lchrusciel) [inline features/product/managing_product_reviews/filtering_product_reviews.feature:11]: I'm lacking information about the state of these reviews or case in which review is rejected

**Review by lchrusciel — APPROVED** [review]

**Pull request merged**